includes/core/classes/class-setup.php

```diff
@@ -292,6 +324,26 @@ public function add_online_event_term(): void {
 		}
 	}
 
+	/**
+	 * Handles actions to be taken when a new site is created in a multisite network.
+	 *
+	 * This function checks if the 'gatherpress' plugin is active across the network.
+	 * If it is, it switches to the new site, calls the `create_table()` function,
+	 * and then restores the current blog.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int $site_id ID of the newly created site.
+	 * @return void
+	 */
+	public function on_site_create( int $site_id ): void {
```

Comment:
This function is hooked onto `wp_initialize_site`, which provides the `WP_Site` object.

```php
	 * @param \WP_Site $site New site object of the created site.
	 * @return void
	 */
	 public function on_site_create( \WP_Site $site ): void {
```